scraper.py

```diff
@@ -114,38 +118,91 @@ def extract_score_data(elem):
+    if test:
         html, encoding = load_inspection_page()
     else:
         html, encoding = get_inspection_page(**kwargs)
-        #code to re-create files as needed
-        # html_part = open('inspection_page.html', 'w')
-        # html_part.write(html)
-        # html_part.closed
-        # encoding_part = open('inspection_page_encoding.html', 'w')
-        # encoding_part.write(encoding)
-        # encoding_part.closed
     doc = parse_source(html, encoding)
     listings = extract_data_listings(doc)
-    final_data = {}
-    for listing in listings:
+    data = []
+    for listing in listings[:count]:
         metadata = extract_restaurant_metadata(listing)
         score_data = extract_score_data(listing)
-        final_data[metadata.get('Business Name')[0]] = score_data
-    report = open('final_info.txt', 'w')
-    for i in final_data.items():
-        report.writelines(str(i))
-        report.writelines('\n')
-    report.closed
-
+        metadata.update(score_data)
+        data.append(metadata)
+
+    if sort == 'highscore':
+        usort = u'High Score'
+    elif sort == 'average':
+        usort = u'Average Score'
+    elif sort == 'inspections':
+        usort = u'Total Inspections'
+
+        data = sorted(data, key=itemgetter(usort),
+                           reverse=(not reverse))
+
+
+    for item in data:
+        yield item
+
+
+def get_geojson(result):
+    address = " ".join(result.get('Address', ''))
+    if not address:
+        return None
+    geocoded = geocoder.google(address)
+    geojson = geocoded.geojson
+    inspection_data = {}
+    use_keys = (
+        'Business Name', 'Average Score', 'Total Inspections', 'High Score',
+        'Address',
+    )
+    for key, val in result.items():
+        if key not in use_keys:
+            continue
+        if isinstance(val, list):
+            val = " ".join(val)
+        inspection_data[key] = val
+    new_address = geojson['properties'].get('address')
+    marker_color = '#FF0000'
+    if new_address:
+        inspection_data['Address'] = new_address
+    geojson['properties'] = inspection_data
+    geojson['properties'].setdefault('marker-color', marker_color)
+    return geojson
 
-    # print listings[0].prettify()
+if __name__ == '__main__':
+    import pprint
+    test = len(sys.argv) > 1 and sys.argv[1] == 'test'
+    total_result = {'type': 'FeatureCollection', 'features': []}
+    parser = argparse.ArgumentParser()
+    parser.add_argument('test',
+                        help="select: test or False",
+                        type=str)
+    parser.add_argument('-s', '--sort',
+                        help="select: average, highscore, inspections",
+                        type=str)
+    parser.add_argument('-c', '--count',
+                        help="enter number of results",
+                        type=int)
+    parser.add_argument('-r', '--reverse',
+                        help="select: True or False",
+                        type=bool)
```

Comment:
I don't think the help text here is strictly true.  IIUC, a boolean flag will contain False if the flag is not provided, and True if it is.